File: 0049-group-anagrams/0049-group-anagrams.java
class Solution {
    public List<List<String>> groupAnagrams(String[] strs) {
        if(strs.length == 0) return new ArrayList();
        Map<String, ArrayList<String>> map = new HashMap<>();
        for(String c : strs){
            char[] ca = c.toCharArray();
            Arrays.sort(ca);
            String key = String.valueOf(ca);
            if(!map.containsKey(key)) map.put(key, new ArrayList());
            map.get(key).add(c);
        }
        return new ArrayList(map.values());
    }
}

File: 0049-group-anagrams/0049-group-anagrams.py
class Solution:
    def groupAnagrams(self, strs: List[str]) -> List[List[str]]:
        out = []
        smap = {}
        for i in strs:
            k = ''.join(sorted(i))
            if k in smap.keys():
                smap[k].append(i)
            else:
                smap[k] = [i]
        for k, v in smap.items():
            out.append(v)
        return out